demo/src/main/java/com/novoda/storagepathfinder/demo/AssetCloner.kt

```diff
@@ -0,0 +1,31 @@
+package com.novoda.storagepathfinder.demo
+
+import android.content.res.AssetManager
+import android.util.Log
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import java.util.concurrent.Executor
+
+internal class AssetCloner(private val assetManager: AssetManager, private val executor: Executor) {
+
+    fun cloneAsset(assetName: String, pathToCloneTo: String) {
+        executor.execute {
+            File(pathToCloneTo)
+                    .also { it.parentFile?.takeIf { !it.exists() }?.also { it.mkdirs() } }
+                    .also { copyAssetToFile(assetName, it) }
+        }
+    }
+
+    private fun copyAssetToFile(assetName: String, outputFile: File) = try {
+        assetManager.open(assetName)
+                .use { input ->
+                    FileOutputStream(outputFile, true).use { output ->
+                        input.copyTo(output)
+                    }
+                }
+    } catch (e: IOException) {
+        Log.e(javaClass.simpleName, "Failed to close streams.", e)
+    }
+
+}
```

Comment:
This is my suggestion 😄 
```
fun cloneAsset(assetName: String, pathToCloneTo: String) {
    executor.execute {
        File(pathToCloneTo)
            .ensurePathIsAvailable()
            .appendAsset(withName = assetName)
    }
}

private fun File.ensurePathIsAvailable(): File = apply {
    parentFile?.takeUnless { it.exists() }?.apply { mkdirs() }
}

private fun File.appendAsset(withName: String) = try {
    assetManager.open(withName).use { input ->
        FileOutputStream(this, true).use { output ->
            input.copyTo(output)
        }
    }
} catch (e: IOException) {
    Log.e(javaClass.simpleName, "Failed to close streams.", e)
}
```
Main thing here is to make the code more readable by using private extensions.

Reply:
💯 I like it